code/game/verbs/suicide.dm

```diff
@@ -8,10 +8,11 @@
 	if(!canSuicide())
 		return
 	if(confirm == "Yes")
-		suiciding = 1
 		var/obj/item/held_item = get_active_hand()
 		if(held_item)
 			var/damagetype = held_item.suicide_act(src)
+			if(damagetype == 0)
+				return //can't kill yourself with that, try something else.
 			if(damagetype)

```

Comment:
you can just replace the three above lines with
```
if(!damagetype)
	return
```
as long as you indent the block below one level down.